docs/content/docs/recipes/audience-react.md

```diff
@@ -0,0 +1,396 @@
+  }, [displayAudience, userId, containerId]);
+
+  const handleContainerNotFound = useCallback(() => {
+    setDisplayAudience(false)
+  }, [setDisplayAudience]);
+```
+
+1. Replace `TODO 2` with the following code. As stated above, the `displayAudience` variable will switch between the `AudienceDisplay` component and `SelectUser` component. Also, functions to update the state variables are passed into components as properties.
+
+```js
+  (displayAudience) ?
+  <AudienceDisplay userId={userId} containerId={containerId} onContainerNotFound={handleContainerNotFound}/> :
+  <SelectUser onSelectUser={handleSelectUser}/>
+```
+
+### Set up AudienceDisplay component
+
+1. Create and open a file `\src\AudienceDisplay.js` in your code editor. Add the following `import` statements:
+
+```js
+  import { useEffect, useState } from "react";
+  import { SharedMap } from "fluid-framework";
+  import { AzureClient } from "@fluidframework/azure-client";
+  import { InsecureTokenProvider } from "@fluidframework/test-client-utils"
+```
+
+1. Add the following functional components and helper functions:
+
+```js
+  const tryGetAudienceObject = async (userId, userName, containerId) => {
+    // TODO 1: Configure the client and container
+    // TODO 2: Configure the container
+    // TODO 3: Get the container and services from the Fluid service.
+    // TODO 4: Return the Audience object.
+  }
+
+  export const AudienceDisplay = (props) => {
+    //TODO 5: Configure user id, user name, and state variables
+    //TODO 6: Set state variables and set event listener on component mount
+    //TODO 7: Return list view
+  }
+
+  const AudienceList = (data) => {
+    //TODO 8: Append view elements to list array for each member
+    //TODO 9: Return list of member elements
+  }
+```
+
+### Getting container and audience
+
+The Fluid runtime will bring changes made to the audience through the container services, but Fluid is agnostic about the UI framework. You can use a helper function to get the Fluid data, from the Audience object into the view layer (the React state). The tryGetAudienceObject method is called when the view component loads after a user id is selected. The returned value is assigned to a React state property.
+
+1. Replace `TODO 1` with the following code. Note that the values for `userId` `userName` `containerId` will be passed in from the `App` component. If there is no `containerId`, a new container is created. Before the client can be used, it needs an `AzureClientProps` that will define the type of connection the client will be using. Think of the `serviceConfig` as the properties required to connect to the service. Note that local mode of Azure Client is used here. To enable collaboration across all client, please replace it with Fluid Relay Service credentials. See how to [set up the Azure Fluid Relay service](https://docs.microsoft.com/en-us/azure/azure-fluid-relay/how-tos/provision-fluid-azure-portal) for more information.
```

Comment:
I don't think the line about FRS should be here. This refers to something a dev would do when they are moving their own app from dev to production. It doesn't belong here in a tutorial where all clients are on the same computer.